controllers/rollingupgrade_controller.go

```diff
@@ -122,6 +131,22 @@ func (r *RollingUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		return ctrl.Result{RequeueAfter: v1alpha1.DefaultRequeueTime}, nil
 	}
 
+	// if the admissionMap is empty, it is possible that the controller was restarted, fetch all running CRs
+	if common.GetSyncMapLen(&r.AdmissionMap) == 0 {
+		r.Info("admission map is empty, fetching all running CRs")
+		var rollingUpgradeList v1alpha1.RollingUpgradeList
+		err := r.List(ctx, &rollingUpgradeList, client.MatchingLabels{v1alpha1.LabelKeyRollingUpgradeCurrentStatus: v1alpha1.StatusRunning})
+		if err != nil {
+			r.Error(err, "failed to fetch running CRs")
+			return ctrl.Result{}, err
+		}
+		for _, cr := range rollingUpgradeList.Items {
+			if cr.CurrentStatus() == v1alpha1.StatusRunning {
```

Comment:
We can probably skip this check if you are only listing the `Running` rolling upgrades.